specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/stable/2020-05-15/timeseriesinsights.json

```diff
@@ -0,0 +1,2549 @@
+        "tags": [
+          "Operations"
+        ],
+        "operationId": "Operations_List",
+        "description": "Lists all of the available Time Series Insights related operations.",
+        "x-ms-examples": {
+          "List available operations for the Time Series Insights resource provider": {
+            "$ref": "./examples/Operation_List.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully listed the available operations.",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.TimeSeriesInsights/environments/{environmentName}": {
+      "put": {
+        "tags": [
+          "Environments"
+        ],
+        "operationId": "Environments_CreateOrUpdate",
+        "x-ms-examples": {
+          "EnvironmentsCreate": {
+            "$ref": "./examples/EnvironmentsCreate.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "description": "Create or update an environment in the specified subscription and resource group.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "environmentName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$",
+            "minLength": 1,
+            "maxLength": 90,
+            "description": "Name of the environment"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EnvironmentCreateOrUpdateParameters"
+            },
+            "description": "Parameters for creating an environment resource."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The existing environment definition was successfully updated.",
+            "schema": {
+              "$ref": "#/definitions/EnvironmentResource"
+            }
+          },
+          "201": {
+            "description": "The environment create request was accepted. Environment provisioning is an asynchronous operation. You can periodically get your environment definition and monitor progress via the provisioningState property.",
+            "schema": {
+              "$ref": "#/definitions/EnvironmentResource"
+            }
+          },
+          "404": {
```

Comment:
>404 [](start = 11, length = 3)

What's the reason for explicitly declaring a 404 response? Most swaggers rely purely on the `default` response.

Reply:
HTTP 404 will happen specifically when either the subscription or resource group, passed as part of the URL - not the request body, is not found.

@sandshadow what do you think of this suggestion, though? If we go for it, then the `default` description becomes misleading instead.

I'd stick with having a separate 404.

Reply:
404 is not the only error that is possible. If you were to go down this way, you would have to document every possible intentional and unintentional HTTP status code. That list would include but is not limited to 400, 403, 404, 429, 500, 502, 503, etc. It is not scalable to have every Swagger include this information when it's common across all RPs. As such the convention is to declare only the `default` error.

On top of that declaring non-`default` error responses has side effects on the SDKs that get generated from this swagger and makes them behave in unexpected ways.